include/aikido/planner/vectorfield/VectorFieldPlanner.hpp

```diff
@@ -2,58 +2,88 @@
+/// A planned trajectroy ends either by a predefined termination criterion or a
+/// integral time.
+///
+/// This class defines two callback functions for a integrator.
+/// step() provides joint velocities the vector field planner should follow,
+/// check() is called after each integration step to check planner status.
+class VectorFieldPlanner
 {
-  TERMINATE,
-  CACHE_AND_TERMINATE,
-  CACHE_AND_CONTINUE,
-  CONTINUE
-};
+public:
+  /// Constructor
+  ///
+  /// \param[in] _vectorField Vector field in configuration space
+  /// \param[in] _constraint Constraint to be satisfied
 
-/// Callback function of joint velocity calculated by a vector field and a
-/// MetaSkeleton.
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _t Planned time of a given duration
-/// \param[out] _dq Joint velocity calculated by a vector field and meta
-/// skeleton
-/// \return Whether vectorfield evaluation succeeds
-using VectorFieldCallback = std::function<bool(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    double _t,
-    Eigen::VectorXd& _dq)>;
+  VectorFieldPlanner(
+      const aikido::planner::vectorfield::ConfigurationSpaceVectorFieldPtr
+          _vectorField,
+      const aikido::constraint::TestablePtr _constraint,
+      double _initialStepSize = 0.1);
 
-/// Callback function of status of planning
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _t Planned time of a given duration
-/// \return Status of vectorfield planner
-using VectorFieldStatusCallback = std::function<VectorFieldPlannerStatus(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    double _t)>;
+  /// Vectorfield callback function that returns joint velocities for
+  /// integration.
+  ///
+  /// \param[in] _q Position in configuration space
+  /// \param[out] _qd Joint velocities in configuration space
+  /// \param[in] _t Current time being planned
+  void step(const Eigen::VectorXd& _q, Eigen::VectorXd& _qd, double _t);
 
-/// Plan to a trajectory by a given vector field.
-///
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in] _constraint Trajectory-wide constraint that must be satisfied
-/// \param[in] _timestep How long an evaluation step is
-/// \param[in] _vectorField Callback of vector field calculation
-/// \param[in] _statusCb Callback of planning status
-/// \return Trajectory or \c nullptr if planning failed
-std::unique_ptr<aikido::trajectory::Spline> planPathByVectorField(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    const aikido::constraint::TestablePtr& _constraint,
-    double _timestep,
-    const VectorFieldCallback& _vectorFieldCb,
-    const VectorFieldStatusCallback& _statusCb);
+  /// Check status after every intergration step
+  ///
+  /// \param[in] _q Position in configuration space
+  /// \param[in] _t Current time being planned
+  void check(const Eigen::VectorXd& _q, double _t);
```

Comment:
I think `step` and `check` functions should be protected because they're just used internally in this class.

Reply:
:+1: